src/marketplace/create_listings.ts

```diff
@@ -1,196 +1,247 @@
+      message: 'Enter the address of the collection contract',
+    })
+  }
 
-      network = await input({
-          message: "Enter the network to be used (mainnet, polygon, etc.)",
-      });
-    }
-  
-    if (!currency) {
-      currency = await input({
-        message:
-          "Enter either a well known currency code (USDC, MATIC, etc.) or a token address",
-      });
-    }
-  
-    if (!price) {
-      price = await input({
-        message: "Enter the price per token, in the currency specified",
-      });
-    }
+  if (tokenId === null || tokenId === undefined) {
+    tokenId = await number({
+      message: 'Enter the ID for the token to be listed',
+    })
+  }
 
-    const chainConfig = findSupportedNetwork(network);
+  if (!network) {
+    console.log(
+      'Please provide the Network for your project as a Sequence chain handle'
+    )
+    console.log(
+      'Possible networks can be found at https://docs.sequence.xyz/solutions/technical-references/chain-support'
+    )
+
+    network = await input({
+      message: 'Enter the network to be used (mainnet, polygon, etc.)',
+    })
+  }
 
-    if (chainConfig === undefined) {
-        program.error("Unsupported network, please select a valid network")
-        return
-    }
+  if (!currency) {
+    currency = await input({
+      message:
+        'Enter either a well known currency code (USDC, MATIC, etc.) or a token address',
+    })
+  }
 
-    let nodeUrl = chainConfig.rpcUrl
+  if (!price) {
+    price = await input({
+      message: 'Enter the price per token, in the currency specified',
+    })
+  }
 
-    if (options.projectAccessKey) {
-        nodeUrl += "/" + options.projectAccessKey
-    }
-    
-    if (options.verbose) {
-      console.log(`Using node URL: ${nodeUrl}`)
-    }
-  
-    const provider = new ethers.JsonRpcProvider(nodeUrl);
-    const wallet = new ethers.Wallet(privateKey, provider);
-  
-    console.log(`Using EOA Wallet: ${wallet.address}`);
-    console.log("Collection Address:", collectionAddress);
-    console.log("Currency token address:", currency);
-  
-    const collectionContract = new ethers.Contract(
-      collectionAddress,
-      ERC1155_ABI,
-      wallet
-    );
-
-    const balance = await collectionContract.balanceOf(wallet.address, tokenId);
-  
-    let isApprovedForAll = false;
-  
+  const chainConfig = findSupportedNetwork(network)
+  if (chainConfig === undefined) {
+    program.error('Unsupported network, please select a valid network')
+    return
+  }
+
+  let nodeUrl = chainConfig.rpcUrl
+
+  if (options.projectAccessKey) {
+    nodeUrl += '/' + options.projectAccessKey
+  }
+
+  if (options.verbose) {
+    console.log(`Using node URL: ${nodeUrl}`)
+  }
+
+  const provider = new ethers.JsonRpcProvider(nodeUrl)
+  const wallet = new ethers.Wallet(privateKey, provider)
+
+  // Set up the wallet client
+  const walletClient = createWalletClient({
+    account: privateKeyToAccount(privateKey),
+    chain: polygon,
```

Comment:
This is polygon only... Lets use the chain config above

Very untested but I think you can get away with something like this:

```suggestion
    chain:  {
			...chainConfig,
			rpcUrls: {
				default: {
					http: [nodeUrl],
				}
			}
		}
```

Reply:
Oh I totally forgot this one.. Thanks for noticing